src/components/schedule-page/table/schedule/schedule-parts/base-row.component.tsx

```diff
@@ -1,121 +1,54 @@
   onBlur?: () => void;
-  resetPointer?: () => void;
+  onDrag?: (pivot: PivotCell, cellInd: number) => void;
+  onDragEnd?: (rowInd: number, cellInd: number) => void;
+  onSave?: (newValue: string) => void;
+  selection?: boolean[];
+  isEditable?: boolean;
 }
 
 export function BaseRowComponentF({
   index,
   dataRow,
   sectionKey,
   cellComponent: CellComponent = BaseCellComponent,
-  showSelectedCells,
   pointerPosition = -1,
   onKeyDown,
   onClick,
-  onRowKeyClick,
   onBlur,
   uuid,
-  resetPointer,
+  onDrag,
+  onDragEnd,
+  onSave,
+  selection = [],
+  isEditable = true,
 }: BaseRowOptions): JSX.Element {
   const scheduleLogic = useContext(ScheduleLogicContext);
-  const [selectedCells, setSelectedCells] = useState<number[]>([]);
-  const [selectionMode, setSelectionMode] = useState(false);
-  const [previousDirectionKey, setPreviousDirectionKey] = useState("");
-
-  useEffect(() => {
-    if (!selectionMode) {
-      setSelectedCells([]);
-    }
-  }, [selectionMode]);
-
   const verboseDates = scheduleLogic?.sections.Metadata?.verboseDates;
-
   const currMonthNumber = scheduleLogic?.sections.Metadata.monthNumber;
-
   const numberOfDays = verboseDates?.length;
 
-  const onContextMenu = useCallback(
-    (cellIndex: number): void => {
-      if (scheduleLogic) {
-        scheduleLogic.changeShiftFrozenState(index, cellIndex);
-      }
-    },
-    [scheduleLogic, index]
-  );
-
   function saveValue(newValue: string): void {
-    if (sectionKey)
-      scheduleLogic?.updateRow(
-        sectionKey,
-        index - 1,
-        [...selectedCells, pointerPosition],
-        newValue
-      );
-    clearSelection();
-  }
-
-  const isFrozen = useCallback((cellInd: number): boolean => {
-    // TODO handle frozen dates
-    return false;
-  }, []);
-
-  function toggleSelection(cellIndex: number): void {
-    if (selectedCells.includes(cellIndex)) {
-      setSelectedCells(
-        selectedCells.filter((selectedCellIndex) => cellIndex !== selectedCellIndex)
-      );
-    } else {
-      setSelectedCells([...selectedCells, cellIndex]);
-    }
-  }
-
-  function handleKeyPress(cellIndex: number, event: React.KeyboardEvent): void {
-    if (event.ctrlKey && DirectionKey[event.key]) {
-      !selectionMode && setSelectionMode(true);
-      if (previousDirectionKey === DirectionKey[event.key] || !selectionMode) {
-        toggleSelection(cellIndex);
-      }
-      setPreviousDirectionKey(DirectionKey[event.key]);
-    } else if (
-      event.key === DirectionKey.ArrowRight ||
-      event.key === DirectionKey.ArrowLeft || // if moves in any direction withour CTRL - disable selection
-      event.key === CellManagementKeys.Escape
-    ) {
-      clearSelection();
-    }
-    onKeyDown && onKeyDown(cellIndex, event);
+    if (sectionKey) onSave && onSave(newValue);
   }
-
-  function clearSelection(): void {
-    resetPointer && resetPointer();
-    setSelectedCells([]);
-    setSelectionMode(false);
-  }
```

Comment:
Ta wszystke logika odpowiadała za wybieranie zakresu komórek za pomocą strzałek. Implementacja trochę się różniła od tego jak to jest zaimplementowane teraz i pomyślałem, że na razie to usunę, ponieważ i tak prawdopodobnie nie będzie używany, a zmiana logiki wymagała by trochę czasu